packages/react-query-devtools/src/utils.ts

```diff
@@ -99,12 +99,18 @@ export function styled<T extends keyof HTMLElementTagNameMap>(
     },
   )
 }
+/**
+ * Resolves to useEffect when "window" is not in scope and useLayoutEffect in the browser
+ */
+export const useIsomorphicEffect = isServer
+  ? React.useEffect
+  : React.useLayoutEffect
 
 export function useIsMounted() {
   const mountedRef = React.useRef(false)
   const isMounted = React.useCallback(() => mountedRef.current, [])
 
-  React[isServer ? 'useEffect' : 'useLayoutEffect'](() => {
+  useIsomorphicEffect(() => {
```

Comment:
especially this should not be needed and have conflicts. we had some troubles with this and changed it to always be `useEffect` only.